core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -34,12 +36,24 @@ abstract class SparkFunSuite
   with Logging {
 // scalastyle:on
 
+  val threadWhiteList = Set(
+    "rpc-client.*", "rpc-server.*", "shuffle-client.*", "shuffle-server.*",
```

Comment:
It would be nice to add comments explaining why the threads are whitelisted. Without an explanation to the contrary, I don't think any of these should be whitelisted.

Reply:
For the new netty related patterns added documentation. Could somebody help me out with rpc and shuffle? All I can see for example TaskSetManagerSuite.test("TaskSet with no preferences") creates a lot of them and I don't see any test issue.

Reply:
Temporarily removed rpc and shuffle. I'll put them back when proper doc can be written.

Reply:
I've made deepdive what these threads are and put documentation for each. I'll execute a build with them and let's see the new numbers.